### ISSUE TITLE
NUOPC does not call cam_final

### ISSUE THREAD

**Issue by MiCurry** [ESCOMP/CAM#411]
When running CAM with the MPAS dycore, messages that are normally outputted when running MCT are not being output. The messages are related to dynamic timings.

The call to write the dynamics timings to the log file is in [`cam_mpas_finalize()`](https://github.com/ESCOMP/CAM/blob/14254c3ec1529108f5c4307f3a22b0cfcb7432d0/src/dynamics/mpas/driver/cam_mpas_subdriver.F90#L2329-L2330). When running MCT, the call tree to `cam_mpas_finalize()` is:

```
atm_final_mct() -> cam_final() -> stepon_final() -> dyn_final() -> cam_mpas_finalize()
```

`atm_comp_nuopc.F90` [imports cam_final](https://github.com/ESCOMP/CAM/blob/14254c3ec1529108f5c4307f3a22b0cfcb7432d0/src/cpl/nuopc/atm_comp_nuopc.F90#L26), but makes no call to it.

Besides the routines to output the dynamics timers, CAM-MPAS has additional cleanup routines that should be called once the run is finished.

**Comment by MiCurry** [comment]
Perhaps a call to `cam_final()` can be made in `ModelFinalize`:

https://github.com/ESCOMP/CAM/blob/14254c3ec1529108f5c4307f3a22b0cfcb7432d0/src/cpl/nuopc/atm_comp_nuopc.F90#L1365-L1393

**Comment by gold2718** [comment]
Nice catch -- and one we should probably add to an upcoming PR (since it is probably just adding the call as you suggest).

**Comment by gold2718** [comment]
Should have been closed by #404
